crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt

```diff
@@ -79,4 +86,91 @@ class Secp256k1Test {
       }
     }
   }
+}
+
+class Web5TestVectorsCryptoEs256k {
+  data class SignTestInput(
+    val data: String,
+    val key: Map<String, Any>?,
+  )
+
+  data class VerifyTestInput(
+    val data: String,
+    val key: Map<String, Any>?,
+    val signature: String,
+  )
+
+  private val mapper = jacksonObjectMapper()
+
+  @Test
+  fun sign() {
+    val typeRef = object : TypeReference<TestVectors<SignTestInput, String>>() {}
+    val testVectors = mapper.readValue(File("../web5-spec/test-vectors/crypto_es256k/sign.json"), typeRef)
+
+    testVectors.vectors.filter { it.errors == false }.forEach { vector ->
+      val inputByteArray: ByteArray = hexStringToByteArray(vector.input.data)
+      val jwkMap = vector.input.key
+      val ecJwk = ECKey.parse(jwkMap.toString())
+      val signedByteArray: ByteArray = Secp256k1.sign(ecJwk, inputByteArray)
+
+      val signedHex = byteArrayToHexString(signedByteArray)
+
+      assertEquals(vector.output, signedHex)
+    }
+
+    testVectors.vectors.filter { it.errors == true }.forEach { vector ->
+      assertFails {
+        val inputByteArray: ByteArray = hexStringToByteArray(vector.input.data)
+        val jwkMap = vector.input.key
+
+        val ecJwk = ECKey.parse(jwkMap.toString())
+
+        Secp256k1.sign(ecJwk, inputByteArray)
+      }
+    }
+  }
+
+  @Test
+  fun verify() {
+    val typeRef = object : TypeReference<TestVectors<VerifyTestInput, Boolean>>() {}
+    val testVectors = mapper.readValue(File("../web5-spec/test-vectors/crypto_es256k/verify.json"), typeRef)
+
+    testVectors.vectors.filter { it.errors == false }.forEach { vector ->
+      val inputByteArray: ByteArray = hexStringToByteArray(vector.input.data)
+      val jwkMap = vector.input.key
+      val signatureByteArray = hexStringToByteArray(vector.input.signature)
+
+      val ecJwk = ECKey.parse(jwkMap.toString())
+
+      if (vector.output == true) {
+        assertDoesNotThrow { Secp256k1.verify(ecJwk, inputByteArray, signatureByteArray)  }
+      } else {
+        assertFails { Secp256k1.verify(ecJwk, inputByteArray, signatureByteArray)  }
+      }
+    }
+
+    testVectors.vectors.filter { it.errors == true }.forEach { vector ->
+      assertFails {
+        val inputByteArray: ByteArray = hexStringToByteArray(vector.input.data)
+        val jwkMap = vector.input.key
+        val signatureByteArray = hexStringToByteArray(vector.input.signature)
+
+        val ecJwk = ECKey.parse(jwkMap.toString())
+        Secp256k1.verify(ecJwk, inputByteArray, signatureByteArray)
+      }
+    }
+  }
+
+  private fun hexStringToByteArray(s: String): ByteArray {
+    val len = s.length
+    val data = ByteArray(len / 2)
+    for (i in 0 until len step 2) {
+      data[i / 2] = ((Character.digit(s[i], 16) shl 4) + Character.digit(s[i + 1], 16)).toByte()
+    }
+    return data
+  }
+
+  private fun byteArrayToHexString(bytes: ByteArray): String {
```

Comment:
Would these be used anywhere else? i.e should they be extracted somewhere shareable?

Reply:
also surprised it's not available via `common`!

Reply:
I changed to use the apache Hex lib